src/templates/contentScript.ts

```diff
@@ -0,0 +1,58 @@
+import { Template } from "./index";
+
+export const ContentScript: Template = {
+  id: "contentScript",
+  name: "Content Script",
+  manifest: {
+    name: "Content Script",
+    description: "Basic Content Script Extension",
+    backgroundScripts: ["background.js"],
+    contentScripts: [
+      {
+        matches: ["<all_urls>"],
+        js: ["content_script.js"],
+      },
+    ],
+  },
+  files: [
+    {
+      name: "background.ts",
+      text: (global) =>
+        `
+${global}.runtime.onInstalled.addListener((details) => {
+  console.log("Extension has been installed. Reason:", details.reason);
+});
+
+function handleMessage(request: any, _sender: chrome.runtime.MessageSender, sendResponse: (response?: any) => void) {
+  console.log("A content script sent a message:", request.greeting);
+  sendResponse({ response: "Response from background script" });
+}
+
+chrome.runtime.onMessage.addListener(handleMessage);
```

Comment:
Note that we hardcode `chrome` here because:

* The `sendResponse` callback param isn't available through the `browser` interface
* There's a years-old Chromium bug open about Chromium not supporting a `Promise` as the return value from the `runtime.onMessage` callback, which otherwise would be what we'd want here: https://bugs.chromium.org/p/chromium/issues/detail?id=1185241

(pls fix)

Reply:
Got it. Definitely wish Chrome supported promises, I've hit that multiple times this week 😅